diff --git a/lib/util.c b/lib/util.c
index 21cc28d..ca45f18 100644
--- a/lib/util.c
+++ b/lib/util.c
@@ -138,6 +138,7 @@ xasprintf(const char *format, ...)
     return s;
 }

+#ifndef HAVE_STRLCPY
 void
 strlcpy(char *dst, const char *src, size_t size)
 {
@@ -148,6 +149,7 @@ strlcpy(char *dst, const char *src, size_t size)
         dst[n_copy] = '\0';
     }
 }
+#endif

 void
 ofp_fatal(int err_no, const char *format, ...)
diff --git a/lib/util.h b/lib/util.h
index fde681f..0f8b063 100644
--- a/lib/util.h
+++ b/lib/util.h
@@ -34,6 +34,7 @@
 #ifndef UTIL_H
 #define UTIL_H 1

+#include <config.h>
 #include <stdarg.h>
 #include <stdbool.h>
 #include <stddef.h>
